sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -603,6 +607,31 @@ void CurlSession::ReadStatusLineAndHeadersFromRawResponse(
   */
 }
 
+void CurlSession::ReadExpected(Context const& context, uint8_t expected)
+{
+  if (this->m_bodyStartInBuffer == -1 || this->m_bodyStartInBuffer == this->m_innerBufferSize)
+  {
+    // end of buffer, pull data from wire
+    this->m_innerBufferSize = m_connection->ReadFromSocket(
+        context, this->m_readBuffer, Details::c_DefaultLibcurlReaderSize);
+    this->m_bodyStartInBuffer = 0;
+  }
+  auto data = this->m_readBuffer[this->m_bodyStartInBuffer];
+  if (data != expected)
+  {
+    throw TransportException(
+        "Un-expected format in HTTP response. Expecting: " + std::to_string(expected)
+        + " but found: " + std::to_string(data));
```

Comment:
The exception message will contain the integer byte value of the character (for example 13, in decimal, for \r). Is that what you wanted?

For example:
`Un-expected format in HTTP response. Expecting: 13 but found: 97`

Reply:
```suggestion
    throw TransportException(
        "Unexpected format in HTTP response. Expecting: " + std::to_string(expected)
        + ", but found: " + std::to_string(data) + ".");
```